File: tests/fixtures/check_then_act/py/catch_write_fixture.py
# Failure bookkeeping in an except block is not the write the check guards.
async def sync(repo, item_id):
    if not repo.has(item_id):
        return
    try:
        await repo.pull(item_id)
    except Exception as error:
        repo.record_error(item_id, error)

File: tests/fixtures/check_then_act/ts/catch_write_fixture.ts
// Failure bookkeeping in a catch block is not the write the check guards.
export async function sync(repo: any, id: string) {
  if (!repo.has(id)) return;
  try {
    await repo.pull(id);
  } catch (error) {
    repo.recordError(id, error);
  }
}
